objects/src/notes/script.rs

```diff
@@ -0,0 +1,28 @@
+use super::{Digest, NoteError};
+use assembly::{parse_program, ProgramAst};
+
+#[derive(Debug, Eq, PartialEq)]
+pub struct NoteScript {
+    hash: Digest,
+    code: ProgramAst,
+}
+
+impl NoteScript {
+    pub fn new(script_src: &str) -> Result<Self, NoteError> {
+        let code = parse_program(script_src).unwrap();
```

Comment:
Better avoid to unwrap on public input. Also, we already return a `Result`, so it's transparent to `?` here

Reply:
This part is still WIP - so, I left the `unwrap` there for now. Will remove once it is flashed out more.